### PR TITLE
[Feature:System] Remove hardcoding of websocket port

### PR DESCRIPTION
**Pull request by MasterOdin**
### What is the current behavior?

The websocket port within Submitty is hardcoded in the code to use `8443`. This caused collisions if trying to use Ubuntu 18.04 and 20.04 VMs at the same time (let alone additional worker machines).

### What is the new behavior?

This reworks the code to allow the websocket to be set to arbitrary values, being set within the `/usr/local/submitty/config/submitty.json` file, and then that's read in when loading the page. If the value is not set, it'll falback to the default value of `8443` like before. 

### Other information?

Documentation for this will be done as part of #6813.

### CODE REVIEW

**Review by williamjallen — APPROVED** [review]
I have reviewed the code twice and tested the branch on my machine.  I have not tested having two VMs running yet but I expect that to work appropriately.

File: site/public/js/server.js [hunk 1]
```diff
@@ -1,14 +1,5 @@
-var csrfToken = undefined;
```
Comment (bmcutler) [inline site/public/js/server.js:1]: is this edit related to this PR?

Comment (MasterOdin) [reply site/public/js/server.js:1]: Not strictly needed, but as I was adding a new variable to be loaded into `window`, I took the opportunity to clean up how these were being set (the `var csrfToken = undefined` is technically doing `window.csrfToken = undefiend`) so that they're not available immediately on page load, instead of having to wait for the onload function below that line to run.
